packages/editor-ui/src/components/RunData.vue

```diff
@@ -895,6 +896,8 @@ export default defineComponent({
 				return name.charAt(0).toLocaleUpperCase() + name.slice(1);
 			}
 			const branches: ITab[] = [];
+			// SwitchV2 allows to create outputs dynamically so we need to check for them
+			const rules = (this.activeNode?.parameters.rules as INodeParameters)?.rules ?? [];
```

Comment:
Could we solve this by doing an expression in `outputNames` in `SwitchV2.node.ts`. Would be nicer if this wasn't node type specific :smile:

Reply:
That's a good point. I've reverted this and instead refactored `getOutputName`to be able to retreive `displayName` from output configuration. Definitely makes it more clear, thanks!